storage: improve message for raft proxying error

tbg opened the pull request:
Saw this a bunch while debugging a test failure and found it lacking descriptiveness.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7797)

<!-- Reviewable:end -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[storage/raft_transport.go, line 138 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7797#-KMZCwkVFiYiZIjrdolt:-KMZCwkVFiYiZIjrdolu:1185192943) ([raw file](https://github.com/cockroachdb/cockroach/blob/b60e1d43401bf36bb3a7811940ef6d7008affd4d/storage/raft_transport.go#L138)):*
> ```Go
> 						return errors.Errorf(
> 							"unable to proxy message to node %d: store %d not registered",
> 							req.Message.To, req.ToReplica.StoreID)
> ```

A lot of raft logging prints the entire replica descriptor (i.e. `req.ToReplica`) via `%s`. Might want to do so here for consistency.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7797#-:-KMZD77huKNgK792E7ZV:1417088721)*
<!-- Sent from Reviewable.io -->

tamird commented on storage/raft_transport.go:
```diff
@@ -133,7 +133,9 @@ func (t *RaftTransport) RaftMessage(stream MultiRaft_RaftMessageServer) (err err
 					if !ok {
-						return errors.Errorf("Unable to proxy message to node: %d", req.Message.To)
+						return errors.Errorf(
+							"unable to proxy message to node %d: store %d not registered",
```
is req.Message.To even a node? I think it's a replica.

arjunravinarayan replied:
It's definitely a replicaID. Use `req.FromReplica.NodeID`.


Arjun Narayan

On Wed, Jul 13, 2016 at 9:48 AM, Tamir Duberstein <notifications@github.com>
wrote:

> In storage/raft_transport.go
> <https://github.com/cockroachdb/cockroach/pull/7797#discussion_r70627613>:
>
> > @@ -133,7 +133,9 @@ func (t *RaftTransport) RaftMessage(stream MultiRaft_RaftMessageServer) (err err
> >  					t.mu.Unlock()
> >
> >  					if !ok {
> > -						return errors.Errorf("Unable to proxy message to node: %d", req.Message.To)
> > +						return errors.Errorf(
> > +							"unable to proxy message to node %d: store %d not registered",
>
> is req.Message.To even a node? I think it's a replica.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/7797/files/b60e1d43401bf36bb3a7811940ef6d7008affd4d#r70627613>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABfkicuyW0qMXMCpJCpYYg8nZMHYsXU6ks5qVOyhgaJpZM4JLWm6>
> .
>

tamird replied:
eh, that's not right either. why are we logging a node anyway? we should log `req.ToReplica`.

tschottdorf replied:
Done.

tbg committed:
storage: improve message for raft proxying error

The new message reads

```
W160714 02:31:26.332774 storage/replica.go:331  range 2: outgoing raft
transport stream to {2 2 2} closed by the remote: rpc error: code = 2 desc
= unable to accept Raft message from {NodeID:2 StoreID:2 ReplicaID:2}: no store
registered for {NodeID:4 StoreID:4 ReplicaID:4}
```

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/raft_transport.go, line 138 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7797#-KMZCwkVFiYiZIjrdolt:-KMc6BAF6Yb2V91lBZew:-1161166686) ([raw file](https://github.com/cockroachdb/cockroach/blob/b60e1d43401bf36bb3a7811940ef6d7008affd4d/storage/raft_transport.go#L138)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> A lot of raft logging prints the entire replica descriptor (i.e. `req.ToReplica`) via `%s`. Might want to do so here for consistency.
</details>
Hmm, that prints `{2 2 2}`. Maybe that would be more consistent, but it looks less helpful. Going to stick with `%+v` for now (`{NodeID:2 StoreID:2 ReplicaID:2}`).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7797)*
<!-- Sent from Reviewable.io -->
